From a3f17b85d5164b7f3cd4cff6b220a0a211cdeba2 Mon Sep 17 00:00:00 2001
From: Chillbo <36623079+Ch1llb0@users.noreply.github.com>
Date: Fri, 15 Feb 2019 01:05:52 +0100
Subject: [PATCH 1/3] "Hide video player OSD during pause" option fix

DialogSeekBar.xml:
- fix conditional visibility (Player.DisplayAfterSeek is also returning true, if player is paused)

MusicVisualisation.xml:
- fix conditional visibility of player status to streamline music and video player OSD behaviour
- add fade animations to player status symbols
- add play symbol for Player.DisplayAfterSeek

VideoFullScreen.xml:
- fix conditional visibilty to show player status also after seeking, but not during seeking
- add fade animations to player status symbols
- add play symbol for Player.DisplayAfterSeek
---
 16x9/DialogSeekBar.xml      |  4 ++--
 16x9/MusicVisualisation.xml | 30 +++++++++++++++++++++++++++++-
 16x9/VideoFullScreen.xml    | 30 +++++++++++++++++++++++++++++-
 3 files changed, 60 insertions(+), 4 deletions(-)

diff --git a/16x9/DialogSeekBar.xml b/16x9/DialogSeekBar.xml
index f41da74..1b20d3e 100755
--- a/16x9/DialogSeekBar.xml
+++ b/16x9/DialogSeekBar.xml
@@ -2,7 +2,7 @@
 <window>
 	<!-- seekbar -->
 	<defaultcontrol always="true">901</defaultcontrol>
-		<visible>[Player.Seeking | Player.DisplayAfterSeek | Player.Forwarding | Player.Rewinding | Player.IsTempo | Player.Paused + String.IsEqual(Skin.String(HideOSD),Always) | Player.Paused + !System.IdleTime(5) + String.IsEqual(Skin.String(HideOSD),5s) | Player.Paused + !System.IdleTime(10) + String.IsEqual(Skin.String(HideOSD),10s) | Player.Paused + !System.IdleTime(20) + String.IsEqual(Skin.String(HideOSD),20s) | Player.Paused + !System.IdleTime(30) + String.IsEqual(Skin.String(HideOSD),30s) | Player.Paused + !System.IdleTime(60) + String.IsEqual(Skin.String(HideOSD),1 min) | Player.Paused + !System.IdleTime(120) + String.IsEqual(Skin.String(HideOSD),2 min) | Player.Paused + !System.IdleTime(180) + String.IsEqual(Skin.String(HideOSD),3 min) | Player.Paused + !System.IdleTime(300) + String.IsEqual(Skin.String(HideOSD),5 min) | !String.IsEmpty(PVR.ChannelNumberInput)] + !Window.IsVisible(subtitlesearch) + !Window.IsVisible(gamevideofilter) + !Window.IsVisible(gamestretchmode) + !Window.IsVisible(gamevideorotation) + !Window.IsVisible(gamevideofilter) + !Window.IsVisible(gameosd)</visible>
+		<visible>[Player.Seeking | Player.DisplayAfterSeek + !Player.Paused | Player.Forwarding | Player.Rewinding | Player.IsTempo | Player.Paused + String.IsEqual(Skin.String(HideOSD),Always) | Player.Paused + !System.IdleTime(5) + String.IsEqual(Skin.String(HideOSD),5s) | Player.Paused + !System.IdleTime(10) + String.IsEqual(Skin.String(HideOSD),10s) | Player.Paused + !System.IdleTime(20) + String.IsEqual(Skin.String(HideOSD),20s) | Player.Paused + !System.IdleTime(30) + String.IsEqual(Skin.String(HideOSD),30s) | Player.Paused + !System.IdleTime(60) + String.IsEqual(Skin.String(HideOSD),1 min) | Player.Paused + !System.IdleTime(120) + String.IsEqual(Skin.String(HideOSD),2 min) | Player.Paused + !System.IdleTime(180) + String.IsEqual(Skin.String(HideOSD),3 min) | Player.Paused + !System.IdleTime(300) + String.IsEqual(Skin.String(HideOSD),5 min) | !String.IsEmpty(PVR.ChannelNumberInput)] + !Window.IsVisible(subtitlesearch) + !Window.IsVisible(gamevideofilter) + !Window.IsVisible(gamestretchmode) + !Window.IsVisible(gamevideorotation) + !Window.IsVisible(gameosd)</visible>
 		<visible>VideoPlayer.IsFullscreen | Window.IsVisible(visualisation)</visible>
 		<zorder>0</zorder>
 
@@ -37,4 +37,4 @@
 
 	</controls>
 
-</window>
+</window>
\ No newline at end of file
diff --git a/16x9/MusicVisualisation.xml b/16x9/MusicVisualisation.xml
index 5df7966..615eac0 100755
--- a/16x9/MusicVisualisation.xml
+++ b/16x9/MusicVisualisation.xml
@@ -126,7 +126,7 @@
 			<control type="group">
 				<width>140</width>
 				<height>60</height>
-				<visible>player.forwarding | player.rewinding | player.paused | player.istempo</visible>
+				<visible>[player.forwarding | player.rewinding | player.istempo | Player.DisplayAfterSeek | player.paused] + !Player.Seeking</visible>
 				<animation type="Visible">
 					<effect type="zoom" start="90" end="100" center="auto" tween="back" easing="out" time="200"/>
 					<effect type="fade" start="0" end="100" time="200"/>
@@ -151,6 +151,13 @@
 					<align>center</align>
 					<label>$VAR[PlayerStatus]</label>
 					<textcolor>$VAR[TextColor2]</textcolor>
+					<visible>player.forwarding | player.rewinding | player.istempo</visible>
+					<animation type="Visible">
+						<effect type="fade" start="0" end="100" time="200"/>
+					</animation>
+					<animation type="Hidden">
+						<effect type="fade" start="100" end="0" time="200"/>
+					</animation>
 				</control>
 
 				<!-- Pause -->
@@ -160,6 +167,27 @@
 					<height>60</height>
 					<texture>osd/OSDPauseNF.png</texture>
 					<visible>player.paused</visible>
+					<animation type="Visible">
+						<effect type="fade" start="0" end="100" time="200"/>
+					</animation>
+					<animation type="Hidden">
+						<effect type="fade" start="100" end="0" time="200"/>
+					</animation>
+				</control>
+				
+				<!-- Play after seek -->
+				<control type="image">
+					<left>40</left>
+					<width>60</width>
+					<height>60</height>
+					<texture colordiffuse="$VAR[DialogColor1]">osd/OSDPlayNF.png</texture>
+					<visible>Player.DisplayAfterSeek + ![Player.Paused | player.forwarding | player.rewinding | player.istempo]</visible>
+					<animation type="Visible">
+						<effect type="fade" start="0" end="100" time="200"/>
+					</animation>
+					<animation type="Hidden">
+						<effect type="fade" start="100" end="0" time="200"/>
+					</animation>
 				</control>
 
 			</control>
diff --git a/16x9/VideoFullScreen.xml b/16x9/VideoFullScreen.xml
index 0cafe55..01cee40 100755
--- a/16x9/VideoFullScreen.xml
+++ b/16x9/VideoFullScreen.xml
@@ -75,7 +75,7 @@
 				<width>140</width>
 				<height>60</height>
 				<visible>!Window.IsVisible(VideoBookmarks)</visible>
-				<visible>[player.forwarding | player.rewinding | player.istempo] + !Player.HasGame | player.paused</visible>
+				<visible>[player.forwarding | player.rewinding | player.istempo | Player.DisplayAfterSeek] + !Player.HasGame + !Player.Seeking | player.paused</visible>
 				<animation type="Visible">
 					<effect type="zoom" start="90" end="100" center="auto" tween="back" easing="out" time="200"/>
 					<effect type="fade" start="0" end="100" time="200"/>
@@ -100,6 +100,13 @@
 					<align>center</align>
 					<label>$VAR[PlayerStatus]</label>
 					<textcolor>$VAR[TextColor2]</textcolor>
+					<visible>player.forwarding | player.rewinding | player.istempo</visible>
+					<animation type="Visible">
+						<effect type="fade" start="0" end="100" time="200"/>
+					</animation>
+					<animation type="Hidden">
+						<effect type="fade" start="100" end="0" time="200"/>
+					</animation>
 				</control>
 
 				<!-- Pause -->
@@ -109,6 +116,27 @@
 					<height>60</height>
 					<texture colordiffuse="$VAR[DialogColor1]">osd/OSDPauseNF.png</texture>
 					<visible>player.paused</visible>
+					<animation type="Visible">
+						<effect type="fade" start="0" end="100" time="200"/>
+					</animation>
+					<animation type="Hidden">
+						<effect type="fade" start="100" end="0" time="200"/>
+					</animation>
+				</control>
+				
+				<!-- Play after seek -->
+				<control type="image" id="1">
+					<left>40</left>
+					<width>60</width>
+					<height>60</height>
+					<texture colordiffuse="$VAR[DialogColor1]">osd/OSDPlayNF.png</texture>
+					<visible>Player.DisplayAfterSeek + ![Player.Paused | player.forwarding | player.rewinding | player.istempo]</visible>
+					<animation type="Visible">
+						<effect type="fade" start="0" end="100" time="200"/>
+					</animation>
+					<animation type="Hidden">
+						<effect type="fade" start="100" end="0" time="200"/>
+					</animation>
 				</control>
 
 			</control>

From af42f87846b864ba13624df62902127dc643c6d9 Mon Sep 17 00:00:00 2001
From: Chillbo <36623079+Ch1llb0@users.noreply.github.com>
Date: Mon, 18 Feb 2019 14:08:42 +0100
Subject: [PATCH 2/3] Add welcome dialog on non-OSMC devices

AddonBrowser.xml:
- add onload to show welcome screen after installing the skin from zip

Custom_Welcom.xml
- add new Welcome dialog for non-OSMC devices

Home.xml:
- add onload to show welcome screen when Home screen first shows up

SettingsCategory.xml:
- add onload to show welcome screen after activating the skin from settings

addon.xml:
- update changelog

Changelog.md:
- update changelog
---
 16x9/AddonBrowser.xml     |  1 +
 16x9/Custom_Welcome.xml   | 66 +++++++++++++++++++++++++++++++++++++++
 16x9/Home.xml             |  1 +
 16x9/SettingsCategory.xml |  1 +
 Changelog.md              |  7 +++++
 addon.xml                 |  2 +-
 6 files changed, 77 insertions(+), 1 deletion(-)
 create mode 100644 16x9/Custom_Welcome.xml

diff --git a/16x9/AddonBrowser.xml b/16x9/AddonBrowser.xml
index 6e395f4..6fe52bd 100755
--- a/16x9/AddonBrowser.xml
+++ b/16x9/AddonBrowser.xml
@@ -4,6 +4,7 @@
 	<backgroundcolor>0x00000000</backgroundcolor>
 	<defaultcontrol always="true">500</defaultcontrol>
 	<views>500</views>
+	<onload condition="String.IsEmpty(Skin.String(WelcomeDialogShown)) + !System.HasAddon(service.osmc.settings)">ActivateWindow(1101)</onload>
 	<onunload condition="System.HasAddon(script.skinshortcuts)">RunScript(script.skinshortcuts,type=buildxml&amp;mainmenuID=9000&amp;group=mainmenu&amp;levels=1&amp;options=noGroups)</onunload>
 	<onload condition="String.IsEmpty(Skin.String(PlotFont))">Skin.SetString(PlotFont,S)</onload>
 	<onload condition="String.IsEmpty(Skin.String(HideOSD))">Skin.SetString(HideOSD,Always)</onload>
diff --git a/16x9/Custom_Welcome.xml b/16x9/Custom_Welcome.xml
new file mode 100644
index 0000000..209c062
--- /dev/null
+++ b/16x9/Custom_Welcome.xml
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<window type="dialog" id="1101">
+	<!-- yesnodialog -->
+	<include>WindowFadeAnimation</include>
+	<defaultcontrol always="true">9001</defaultcontrol>
+	<controls>
+
+		<!-- Background -->
+		<include>WindowFadeBackgroundImage</include>
+
+		<include content="Time">
+			<param name="heading">Welcome</param>
+		</include>
+
+		<!-- Heading -->
+		<control type="label" id="1">
+			<left>-15000</left>
+			<top>40</top>
+			<width>600</width>
+			<height>48</height>
+			<align>center</align>
+			<font>Font48</font>
+			<textcolor>$VAR[TextColor1]</textcolor>
+		</control>
+
+		<!-- Details -->
+		<control type="textbox" id="9">
+			<centerleft>50%</centerleft>
+			<centertop>50%</centertop>
+			<width>1880</width>
+			<height>600</height>
+			<align>center</align>
+			<aligny>center</aligny>
+			<font>Font36</font>
+			<autoscroll>False</autoscroll>
+			<textcolor>$VAR[TextColor1]</textcolor>
+			<label>We'd like to thank you for using the OSMC skin on your device![CR]For more information see https://osmc.tv ...</label>
+		</control>
+
+		<!-- Button grouplist background -->
+		<include content="dialogButtonBackground">
+			<param name="id">9001</param>
+		</include>
+
+		<!-- Buttons -->
+		<control type="grouplist" id="9001">
+			<left>120</left>
+			<bottom>100</bottom>
+			<width>1680</width>
+			<height>66</height>
+			<orientation>horizontal</orientation>
+			<align>center</align>
+			<itemgap>30</itemgap>
+			<control type="button">
+				<width>auto</width>
+				<font>Font36</font>
+				<align>center</align>
+				<onclick>Dialog.Close(1101)</onclick>
+				<onclick>Skin.SetString(WelcomeDialogShown,true)</onclick>
+				<label>OK</label>
+			</control>
+		</control>
+
+	</controls>
+
+</window>
\ No newline at end of file
diff --git a/16x9/Home.xml b/16x9/Home.xml
index 557ff5b..f4b48ec 100755
--- a/16x9/Home.xml
+++ b/16x9/Home.xml
@@ -21,6 +21,7 @@
 	<onload condition="String.IsEmpty(Skin.String(NFDimOpac))">Skin.SetString(NFDimOpac,100)</onload>
 	
 	<!-- Skin settings background + colors -->
+	<onload condition="String.IsEmpty(Skin.String(WelcomeDialogShown)) + !System.HasAddon(service.osmc.settings)">ActivateWindow(1101)</onload>
 	<onload condition="String.IsEqual(Skin.String(BackgroundImage),custom)">Skin.SetString(BackgroundImage,1)</onload>
 	<onload condition="String.IsEmpty(Skin.String(BackgroundImage))">Skin.SetString(BackgroundImage,1)</onload>
 	<onload condition="String.IsEmpty(Skin.String(BackgroundDefaultImage))">Skin.SetString(BackgroundDefaultImage,yes)</onload>
diff --git a/16x9/SettingsCategory.xml b/16x9/SettingsCategory.xml
index 833da03..3da0ef2 100755
--- a/16x9/SettingsCategory.xml
+++ b/16x9/SettingsCategory.xml
@@ -2,6 +2,7 @@
 <window>
 	<!-- settingscategory -->
 	<backgroundcolor>0x00000000</backgroundcolor>
+	<onload condition="String.IsEmpty(Skin.String(WelcomeDialogShown)) + !System.HasAddon(service.osmc.settings)">ActivateWindow(1101)</onload>
 	<onunload condition="System.HasAddon(script.skinshortcuts)">RunScript(script.skinshortcuts,type=buildxml&amp;mainmenuID=9000&amp;group=mainmenu&amp;levels=1&amp;options=noGroups)</onunload>
 	<onload condition="String.IsEmpty(Skin.String(PlotFont))">Skin.SetString(PlotFont,S)</onload>
 	<onload condition="String.IsEmpty(Skin.String(HideOSD))">Skin.SetString(HideOSD,Always)</onload>
diff --git a/Changelog.md b/Changelog.md
index a9fa02f..0f35f4a 100644
--- a/Changelog.md
+++ b/Changelog.md
@@ -9,6 +9,7 @@ _New_
 - new color options (color sets, background gradients, adjustable opacity)
 - add PVR channel number input dialog
 - add PVR timeshift status dialog
+- ddd welcome dialog on non-OSMC devices
 
 _Improved_
 - adjust syntax, values, labels and infobools to match v18 requirements
@@ -62,10 +63,14 @@ _Add new dependency button in addon info dialog to match v18 requirements:_
 Addon.Browser.xml:
 - replace colors by new color variables
 - fix highlighting in focusedlayout of grouplists
+- add onload to show welcome screen after installing the skin from zip
 
 Custom_Cache_Progress.xml:
 - replace colors by new color variables
 
+Custom_Welcom.xml
+- add new Welcome dialog for non-OSMC devices
+
 Defaults.xml:
 - replace colors by new color variables
 - replace focus variable for button highlighting
@@ -222,6 +227,7 @@ GameOSD.xml:
 Home.xml:
 - add new onloads
 - replace colors by new color variables
+- add onload to show welcome screen when Home screen first shows up
 
 Include_DialogSettings.xml:
 - replace colors by new color variables
@@ -326,6 +332,7 @@ Settings.xml:
 SettingsCategory.xml:
 - replace colors by new color variables
 - replace focus variable for button highlighting
+- add onload to show welcome screen after activating the skin from settings
 
 SettingsProfile.xml:
 - replace colors by new color variables
diff --git a/addon.xml b/addon.xml
index b83db11..510e53f 100644
--- a/addon.xml
+++ b/addon.xml
@@ -14,7 +14,7 @@
 		<platform>all</platform>
 		<summary lang="en">The default skin for OSMC</summary>
 		<description lang="en">Original skin: Andy Morton[CR]Original design: Simon Brunton[CR]Skinner: Julian Michel</description>
-		<news>OSMC Kodi v18 update[CR][CR][B]New[/B][CR]- add new v18 subtitle settings OSD during fullscreen video playback[CR]- add new games section to match v18 requirements[CR]- add games section[CR]- add new resolution select button/dialog in video player[CR]- add player icon to now playing dialog[CR]- add new dependency button in addon info dialog to match v18 requirements[CR]- new color options (color sets, background gradients, adjustable opacity)[CR]- add PVR channel number input dialog[CR]- add PVR timeshift status dialog[CR][CR][B]Improved[/B][CR]- adjust syntax, values, labels and infobools to match v18 requirements[CR]- adjust PVR section to match v18 requirements[CR]- highlighting color now adjusts according to text color[CR]- streamline OSD animations[CR]- add missing adjustable plot fonts[CR]- let favourites dialog behave like a normal window[CR][CR][B]Fixed[/B][CR]- show proper game widget title when not using skinshortcuts script[CR]- highlighting is now more consistent[CR]- fix current position/time remaining and current time/end time for PVR playback[CR]- hide deprecated previous/next channel buttons in PVR playback OSD[CR]- fix background of subtitle settings window[CR]- fix layout of PVR playback dialogs</news>
+		<news>OSMC Kodi v18 update[CR][CR][B]New[/B][CR]- add new v18 subtitle settings OSD during fullscreen video playback[CR]- add new games section to match v18 requirements[CR]- add games section[CR]- add new resolution select button/dialog in video player[CR]- add player icon to now playing dialog[CR]- add new dependency button in addon info dialog to match v18 requirements[CR]- new color options (color sets, background gradients, adjustable opacity)[CR]- add PVR channel number input dialog[CR]- add PVR timeshift status dialog[CR]- add welcome dialog on non-OSMC devices[CR][CR][B]Improved[/B][CR]- adjust syntax, values, labels and infobools to match v18 requirements[CR]- adjust PVR section to match v18 requirements[CR]- highlighting color now adjusts according to text color[CR]- streamline OSD animations[CR]- add missing adjustable plot fonts[CR]- let favourites dialog behave like a normal window[CR][CR][B]Fixed[/B][CR]- show proper game widget title when not using skinshortcuts script[CR]- highlighting is now more consistent[CR]- fix current position/time remaining and current time/end time for PVR playback[CR]- hide deprecated previous/next channel buttons in PVR playback OSD[CR]- fix background of subtitle settings window[CR]- fix layout of PVR playback dialogs</news>
 	</extension>
 	<assets>
 		<icon>icon.png</icon>

From af98b4ec6aef3050965aee6563cb261c58593f4a Mon Sep 17 00:00:00 2001
From: "Ch1llb0/skin.osmc" <36623079+Ch1llb0@users.noreply.github.com>
Date: Mon, 18 Feb 2019 17:57:35 +0100
Subject: [PATCH 3/3] PVR Channel number input bugfix

DialogPVRChannelGuide.xml:
- set zorder 0 to show PVR channel number input dialog on top

DialogPVRChannelsOSD.xml:
- set zorder 0 to show PVR channel number input dialog on top
---
 16x9/DialogPVRChannelGuide.xml | 1 +
 16x9/DialogPVRChannelsOSD.xml  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/16x9/DialogPVRChannelGuide.xml b/16x9/DialogPVRChannelGuide.xml
index b59d48c..3457c3c 100644
--- a/16x9/DialogPVRChannelGuide.xml
+++ b/16x9/DialogPVRChannelGuide.xml
@@ -3,6 +3,7 @@
 	<!-- osdvideosettings -->
 	<defaultcontrol always="true">11</defaultcontrol>
 	<include>DialogZoomAnimation</include>
+	<zorder>0</zorder>
 	
 	<controls>
 
diff --git a/16x9/DialogPVRChannelsOSD.xml b/16x9/DialogPVRChannelsOSD.xml
index 2c60987..fe79a37 100755
--- a/16x9/DialogPVRChannelsOSD.xml
+++ b/16x9/DialogPVRChannelsOSD.xml
@@ -3,6 +3,7 @@
 	<!-- pvrosdchannels -->
 	<defaultcontrol always="true">11</defaultcontrol>
 	<include>DialogZoomAnimation</include>
+	<zorder>0</zorder>
 
 	<controls>
 
